app/Helpers/DashboardHelper.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  Created by Backbase on 13/02/2024.
+//
+
+import BackbaseDesignSystem
+import UIKit
+import AccountsJourney
+import GoldenAccountsUseCase
+import Resolver
+import UserManagerUserProfileUseCase
+import UserProfileJourney
+
+private struct UserPresentable: TabHeaderViewControllerUserPresentable {
+    var name: String
+    var company: String?
+    var image: UIImage?
+}
+
+/// A helper struct to setup the dashboard.
+struct DashboardHelper {
+    func getViewController(navigationController: UINavigationController, serviceAgreementName: String) async -> UIViewController {
+        let userProfileName = await fetchUserProfile()
+        return DispatchQueue.main.sync {
+            getTabHeaderViewController(navigationController, userProfileName, serviceAgreementName)
+        }
+    }
+
+    private func fetchUserProfile() async -> String {
+        let userProfileUseCase: UserProfileUseCase = Resolver.resolve()
+        return await withCheckedContinuation { continuation in
+            DispatchQueue.global().async {
+                userProfileUseCase.retrieveUserProfile { result in
+                    switch result {
+                    case .success(let userProfile):
+                        continuation.resume(returning: userProfile.fullName)
+                    case .failure:
+                        continuation.resume(returning: "") // This is not a bug, but a feature.
+                    }
+                }
+            }
+        }
+    }
+
+    private func getTabHeaderViewController(_ navigationController: UINavigationController,
+                                            _ userName: String,
+                                            _ serviceAgreementName: String) -> UIViewController {
+        let accountsListViewController = AccountsList.build(navigationController: navigationController)
+        accountsListViewController.title = Bundle.main.localize("accountsJourney.accountsList.labels.title") ?? ""
+        let tab2ViewController = ComingSoonViewController(title: Bundle.main.localize("dashboard.menu.tab2") ?? "")
+        let tab3ViewController = ComingSoonViewController(title: Bundle.main.localize("dashboard.menu.tab3") ?? "")
```

Comment:
Seeing that these are just placeholders, i think we can maybe add one tab instead of two. Or give it a meaningful name or something.

Reply:
@Emils comment: The idea would be to show multiple tabs as a sample, maybe naming them "Sample tab" would help that a bit?

Reply:
I don't know, but i think it'll be better is there's an alignment first between both Android and iOS to ensure consistency, also this can provide a baseline for future additions

Reply:
this already aligns with android actually. https://github.com/Backbase/golden-sample-app-android/pull/15